### PR TITLE
[Move] Add type immunity removal moves

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes?
Implement Foresight, Miracle Eye, and Odor Sleuth. 

## Why am I doing these changes?
Implement more moves.

## What did change?
The moves Foresight, Miracle Eye, and Odor Sleuth are implemented by applying `BattlerTag`s that modify the type immunity check and evasion level check.

## How to test the changes?
Use Foresight against a Ghost type Pokémon and then use a Normal or Fighting type attacking move against it.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [ ] There is no overlap with another PR?
  - Updated version of #812
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

File: src/data/battler-tags.ts [hunk 1]
```diff
@@ -1647,6 +1647,47 @@ export class StockpilingTag extends BattlerTag {
...
+ * Tag that makes the target drop all of it type immunities
+ * and all accuracy checks ignore its evasiveness stat.
+ *
+ * Applied by moves: {@linkcode Moves.ODOR_SLEUTH | Odor Sleuth},
+ * {@linkcode Moves.MIRACLE_EYE | Miracle Eye} and {@linkcode Moves.FORESIGHT | Foresight}.
+ *
+ * @extends BattlerTag
+ * @see {@linkcode ignoreImmunity}
+ */
+export class ExposedTag extends BattlerTag {
```
Comment (Tempo-anon) [inline src/data/battler-tags.ts:1660]: Odor Sleuth, Foresight, and Miracle Eye also add a secondary effect on the Pokemon in that all attacks aimed at them will have -1 accuracy (always hit) and ignores any accuracy stat buffs (not sure if that matters if all moves aimed at them will become -1 accuracy)

File: src/locales/fr/move-trigger.ts [hunk 2]
```diff
@@ -59,4 +59,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "copyType": "{{pokemonName}} prend le type\nde {{targetPokemonName}} !",
   "suppressAbilities": "Le talent de {{pokemonName}}\na été rendu inactif !",
   "swapArenaTags": "Les effets affectant chaque côté du terrain\nont été échangés par {{pokemonName}} !",
+  "exposedMove": "{{pokemonName}} identified\n{{targetPokemonName}}!",
```
Comment (Adri1) [inline src/locales/fr/move-trigger.ts:62]:
```suggestion
  "exposedMove": "{{targetPokemonName}} est identifié\npar {{pokemonName}} !",
```

File: src/locales/de/move-trigger.ts [hunk 3]
```diff
@@ -59,4 +59,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "copyType": "{{pokemonName}} hat den Typ von {{targetPokemonName}} angenommen!",
   "suppressAbilities": "Die Fähigkeit von {{pokemonName}} wirkt nicht mehr!",
   "swapArenaTags": "{{pokemonName}} hat die Effekte, die auf den beiden Seiten des Kampffeldes wirken, miteinander getauscht!",
+  "exposedMove": "{{pokemonName}} identified\n{{targetPokemonName}}!",
```
Comment (CodeTappert) [inline src/locales/de/move-trigger.ts:62]:
```suggestion
  "exposedMove": "{{pokemonName}} erkennt {{targetPokemonName}}!",
```

File: src/data/battler-tags.ts (hunk 1 above)
Comment (DayKev) [reply src/data/battler-tags.ts:1660]: Wait I misread that, it's only that Foresight (et al) have `-1` (always hit) accuracy. Okay, crisis averted, everything makes sense again.

File: src/locales/pt_BR/move-trigger.ts [hunk 4]
```diff
@@ -59,4 +59,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "copyType": "O tipo de {{pokemonName}}\nmudou para combinar com {{targetPokemonName}}!",
   "suppressAbilities": "A habilidade de {{pokemonName}}\nfoi suprimida!",
   "swapArenaTags": "{{pokemonName}} trocou os efeitos de batalha que afetam cada lado do campo!",
+  "exposedMove": "{{pokemonName}} identified\n{{targetPokemonName}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/move-trigger.ts:62]:
```suggestion
  "exposedMove": "{{pokemonName}} identificou\n{{targetPokemonName}}!",
```

File: src/locales/zh_TW/move-trigger.ts [hunk 5]
```diff
@@ -59,4 +59,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "copyType": "{{pokemonName}}變成了{{targetPokemonName}}的屬性！",
   "suppressAbilities": "{{pokemonName}}的特性\n變得無效了！",
   "swapArenaTags": "{{pokemonName}}\n交換了雙方的場地效果！",
+  "exposedMove": "{{pokemonName}} identified\n{{targetPokemonName}}!",
```
Comment (mercurius-00) [inline src/locales/zh_TW/move-trigger.ts:62]:
```suggestion
  "exposedMove": "{{pokemonName}}識破了\n{{targetPokemonName}}的原形！",
```

File: src/locales/zh_CN/move-trigger.ts [hunk 6]
```diff
@@ -59,4 +59,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "copyType": "{{pokemonName}}\n变成了{{targetPokemonName}}的属性！",
   "suppressAbilities": "{{pokemonName}}的特性\n变得无效了！",
   "swapArenaTags": "{{pokemonName}}\n交换了双方的场地效果！",
+  "exposedMove": "{{pokemonName}} identified\n{{targetPokemonName}}!",
```
Comment (mercurius-00) [inline src/locales/zh_CN/move-trigger.ts:62]:
```suggestion
  "exposedMove": "{{pokemonName}}识破了\n{{targetPokemonName}}的原型！",
```

**Review by mercurius-00 — COMMENTED** [review]
zh

**Comment by EnochG1** [comment]
@DayKev 
From now Japanese is added in beta.
Could you merge latest beta and add Japanese locale files? 🙏

**Pull request merged**